### PR TITLE
Kernel: NVMe UBSAN fix + Misc improvements to NVMe

### PR DESCRIPTION
**Pull request by Panky-codes**
The first commit fixes the UBSAN issue that is referenced here #18870 due to GCC 13.1 upgrade.

The next 4 commits are a general improvement to `NVMe{Poll|Interrupt}Queue` classes which adds a `try_create` method and moves the initialization out of the object construction to avoid any race condition in the future.

### CODE REVIEW

File: Kernel/Storage/NVMe/NVMeQueue.cpp [hunk 1]
```diff
@@ -18,10 +18,10 @@ ErrorOr<NonnullLockRefPtr<NVMeQueue>> NVMeQueue::try_create(NVMeController& devi
     RefPtr<Memory::PhysicalPage> rw_dma_page;
     auto rw_dma_region = TRY(MM.allocate_dma_buffer_page("NVMe Queue Read/Write DMA"sv, Memory::Region::Access::ReadWrite, rw_dma_page));
     if (queue_type == QueueType::Polled) {
-        auto queue = TRY(adopt_nonnull_lock_ref_or_enomem(new (nothrow) NVMePollQueue(move(rw_dma_region), *rw_dma_page, qid, q_depth, move(cq_dma_region), cq_dma_page, move(sq_dma_region), sq_dma_page, move(db_regs))));
+        auto queue = TRY(NVMePollQueue::try_create(move(rw_dma_region), rw_dma_page, qid, q_depth, move(cq_dma_region), cq_dma_page, move(sq_dma_region), sq_dma_page, move(db_regs)));
```
Comment (ADKaster) [inline Kernel/Storage/NVMe/NVMeQueue.cpp:21]: It seems to me like it's an error if the rw_dma_page is not filled in with an actual allocated page? If that's the case we should call `release_nonull()` to pass a NonnullRefPtr to the factory function here.

File: Kernel/Storage/NVMe/NVMeInterruptQueue.cpp [hunk 2]
```diff
@@ -11,6 +11,11 @@
 
 namespace Kernel {
 
+ErrorOr<NonnullLockRefPtr<NVMeInterruptQueue>> NVMeInterruptQueue::try_create(PCI::Device& device, NonnullOwnPtr<Memory::Region> rw_dma_region, RefPtr<Memory::PhysicalPage>& rw_dma_page, u16 qid, u8 irq, u32 q_depth, OwnPtr<Memory::Region> cq_dma_region, Vector<NonnullRefPtr<Memory::PhysicalPage>> cq_dma_page, OwnPtr<Memory::Region> sq_dma_region, Vector<NonnullRefPtr<Memory::PhysicalPage>> sq_dma_page, Memory::TypedMapping<DoorbellRegister volatile> db_regs)
```
[the remark this answers is not in the document]
Comment (Panky-codes) [reply Kernel/Storage/NVMe/NVMeInterruptQueue.cpp:14]: It has been in my todo to do this as it was getting hard to keep track of the parameter list. As we are on the topic of doing misc cleanups, I will do this as well. Thanks :)

[the remark this answers is not in the document]
Comment (Panky-codes) [reply Kernel/Storage/NVMe/NVMeInterruptQueue.cpp:14]: Actually, I realized that we don't need `Vector<NonnullRefPtr<Memory::PhysicalPage>> cq_dma_page` and `Vector<NonnullRefPtr<Memory::PhysicalPage>> sq_dma_page`.  Removing them makes the arguments list a bit more manageable IMO without resorting to a struct pattern.

**Comment by Panky-codes** [comment]
Changes since v1:
- Remove unused {cq|sq}_dma_page from NVMeQueue. This makes the argument list shorter and more readable.
- Convert rw_dma_page to NonnullRefPtr in NVMeQueue. `release_nonnull` is used to pass the `NonnullRefPtr`

Thanks @ADKaster for the comments.

**Review by ADKaster — APPROVED** [review]
Looks good, thanks!

**Pull request merged**